util/expensivequery/memory_usage_alarm.go

```diff
@@ -110,23 +110,35 @@ func (record *memoryUsageAlarm) alarm4ExcessiveMemUsage(sm util.SessionManager)
 		interval := time.Since(record.lastCheckTime)
 		record.lastCheckTime = time.Now()
 		if interval > 10*time.Second {
-			record.doRecord(memoryUsage, sm)
+			record.doRecord(memoryUsage, instanceStats.HeapAlloc, sm)
 		}
 	}
 }
 
-func (record *memoryUsageAlarm) doRecord(memUsage uint64, sm util.SessionManager) {
-	logutil.BgLogger().Warn("the TiDB instance now takes a lot of memory, has the risk of OOM",
-		zap.Bool("is server-momory-quota set", record.isServerMemoryQuotaSet),
-		zap.Any("memory size", record.serverMemoryQuota),
-		zap.Any("memory usage", memUsage),
-		zap.Any("memory-usage-alarm-ratio", config.GetGlobalConfig().Performance.MemoryUsageAlarmRatio),
-	)
+func (record *memoryUsageAlarm) doRecord(memUsage uint64, instanceMemoryUsage uint64, sm util.SessionManager) {
+	if record.isServerMemoryQuotaSet {
```

Comment:
fields = make([]Field, len)
fileds = append(fields, zap.Bool())
if record.isServerMemoryQuotaSet {

}else{
}

logutil.BgLogger().Warn("tidb-server has the risk of OOM. Running SQLs and heap profile will be recorded in `record path`", fields...)